halcyon/hype_detector.py

```diff
@@ -0,0 +1,76 @@
+import regex as re
+
+import emoji
+import numpy as np
+
+from halcyon.scorer import TextScorer
+
+MAX_HYPE_ELEM_CT = 15
+
+
+class HypeDetector(TextScorer):
+    """
+    The HypeDetector uses ML to detect hype in online content
+    """
+
+    def __init__(self, model: object):
+        self.model = model[0]
+        self.vectorizer = model[1]
+        self.labels = ["not_hype", "hype"]
+
+    def _normalize(self, value: int, min_value: int, max_value: int) -> float:
+        """
+        Normalize a value to the range [0, 1].
+        """
+
+        if value < min_value:
+            return 0
+        if value > max_value:
+            return 1
+        return (value - min_value) / (max_value - min_value)
+
+    def clean_text(self, text: str) -> str:
+        """
+        Strip out common text
+        """
+        all_patterns = [
+            "…see more",
+            "https://lnkd.in/",
+            "www.bit.ly/",
+            "http://bit.ly",
+            "#",
+        ]
+        regex_pattern = r"|".join(map(r"(?:{})".format, all_patterns))
+        cleaned_text = re.sub(regex_pattern, "", text)
+        return cleaned_text
+
+    def _hype_score(self, text: str) -> float:
+        """
+        Returns a score in the range [0, 1] that is a representation of
+        the number of hashtags, exclamation points, and emojis used in
+        a post.  This scores gives further insight into how much a post
+        is hyped up by the use of hashtags, emojis, and exclamation points
+        to capture attention.
+        """
+        ht_count = text.count("#")
+        exclamation_count = text.count("!")
+        emoji_count = emoji.emoji_count(text)
```

Comment:
Nice, the emoji count is a good idea and it's great that there's a package for it.